packages/component/src/types/internal/TypeOfArray.ts

```diff
@@ -0,0 +1,7 @@
+export type TypeOfArray<T extends unknown[] | Array<unknown> | ReadonlyArray<unknown>> = T extends (infer I)[]
```

Comment:
I don't think you need to check `T extends (infer I)[]` and `Array<infer I>`. They are the same thing